test/appium/tests/atomic/test_public_chat_browsing.py

```diff
@@ -509,6 +510,453 @@ def test_browser_web3_permissions_testdapp(self):
         self.errors.verify_no_errors()
 
 
+@pytest.mark.xdist_group(name="browser_profile_medium_1")
+@marks.medium
+class TestBrowserProfileOneDevice(MultipleSharedDeviceTestCase):
+
+    @classmethod
+    def setup_class(cls):
+        cls.drivers, cls.loop = create_shared_drivers(1)
+        cls.sign_in = SignInView(cls.drivers[0])
+        cls.home = cls.sign_in.create_user()
+        cls.wiki_texts = ['Español', '日本語', 'Français', '中文', 'Português']
+
+    @marks.testrail_id(702149)
+    def test_browser_can_access_images_by_link(self):
+        urls = {
+            'https://cdn.dribbble.com/users/45534/screenshots/3142450/logo_dribbble.png':
+                'url1.png',
+            'https://steemitimages.com/DQmYEjeBuAKVRa3b3ZqwLicSHaPUm7WFtQqohGaZdA9ghjx/images%20(4).jpeg':
+                'url3.png'
+        }
+        dapp = self.home.dapp_tab_button.click()
+        from views.web_views.base_web_view import BaseWebView
+        base_web = BaseWebView(self.drivers[0])
+        for url in urls:
+            dapp.open_url(url)
+            # self.drivers.set_clipboard_text(url)
+            # dapp.enter_url_editbox.click()
+            # dapp.paste_text()
+            # dapp.confirm()
+            # base_web.wait_for_d_aap_to_load(20)
+
+            if dapp.web_page.is_element_differs_from_template(urls[url], 5):
+                self.errors.append('Web page does not match expected template %s' % urls[url])
+            base_web.browser_previous_page_button.click_until_presence_of_element(dapp.element_by_text_part('Discover'))
+        self.errors.verify_no_errors()
+
+    @marks.testrail_id(702150)
+    def test_browser_back_forward_navigation_history_kept_after_relogin(self):
+
+        dapp = self.home.dapp_tab_button.click()
+        ua_url = 'https://uk.m.wikipedia.org/'
+        browsing = dapp.open_url(ua_url)
+        browsing.element_by_text_part('Ласкаво просимо').wait_for_element(20)
+
+        browsing.just_fyi("Check next page")
+        browsing.just_fyi('Navigate to next page and back')
+        browsing.element_by_text_part('може редагувати кожен').click()
+        browsing.element_by_text_part('написана спільно її читачами').wait_for_element(30)
+        browsing.browser_previous_page_button.click()
+        browsing.wait_for_element_starts_with_text('може редагувати кожен')
+
+        browsing.just_fyi('Relogin and check that tap on "Next" navigates to next page')
+        profile = self.home.profile_button.click()
+        # profile.switch_network()
+        profile.relogin()
```

Comment:
profile.reopen_app() works faster, better to use it if re-login is not required explicitely

Reply:
done